urlcaef331f4e_dreamable_cra_tgz-pJ8-coho_lp_project_LPProjectJ8/src/coho/lp/project/DoubleIntervalLPProject.java

```diff
@@ -0,0 +1,308 @@
+								Matrix newb = A.convert(b.length()+1,1);
+								newb.assign(b,0,0);
+								newb.assign(newCost,b.length());
+								if(DEBUG.debug) DEBUG.println("LPproject.findPlanes(): Insert an constraint "+x.sub(y)+""+newCost);
+								hullAsPlanes = findPlanes(newA,newb,x_axis,y_axis);
+								//remove the edge from the result and return
+								Line delP = new Line(newCost,quad-1,1);
+								for(int i=0;i<hullAsPlanes.size();i++){
+									Line curr = hullAsPlanes.get(i);
+									if(delP.similar(curr)){
+										hullAsPlanes.remove(curr);
+										if(DEBUG.debug) DEBUG.println("LPproject.findPlanes(): Remove the constraint: "+curr.toString());
+										break;
+									}
+								}
+								return hullAsPlanes;
+							}
+						}
+						B = newB;
+						jumpQuads = 0;
+					}else{//next edge in the next quad. Backtrack once for efficient.
+						//if(DEBUG.debug) DEBUG.println(LPproject_DEBUG,"LPproject.findPlanes(): nextNorm() return 'go to next quad', recheck the decision it made");
+						// backtrack till find the minimum angle.						
+						LPBasis newB = lpSolver.opt((x.add(y.mult(maxInc)))).optBasis();//direction=x+y*tan(maxAngle)
+						if(!newB.equals(B)){
+							DoubleIntervalMatrix point = (DoubleIntervalCohoMatrix.create(A.row(B.basis()),false)).getSolution(b.V(B.basis()));								
+							DoubleIntervalMatrix newPoint = (DoubleIntervalCohoMatrix.create(A.row(newB.basis()),false)).getSolution(b.V(newB.basis()));
+							double length =newPoint.sub(point).norm().doubleValue(); 
+							//we may skip edge shorter than minLength
+							if(length>minLength){//should not jump to next quad
+								//if(DEBUG.debug) DEBUG.println(LPproject_DEBUG,"LPproject.findPlanes(): There is a new basis "+newB.toString()+" in the same quadrant.");
+								Iterator<LPBasis> iter = result.optBasesIter();
+								try{
+									while(!B.equals(iter.next())){
+									}//until we find the current basis
+									if(iter.hasNext()){//if contains the right endpoint
+										//backtrackTimes=0;//reset
+										B = iter.next();//use the next basis										
+										//if(DEBUG.debug) DEBUG.println(LPproject_DEBUG,"LPproject.findPlanes(): Let's try another possible optimal basis "+B.toString());
+									}else{//force to the right endpoint
+										//XXX: here, we don't update alpha now because we update backtrackTimes
+										//result = rightEnd(alpha,currNorm.E(),B,result,solver,A,x,y,false,true,8*backtrackTimes);//last para can be true or false. result.bestOptBasis==B
+										//at least once, at most 5 times.
+										result = rightEnd(alpha,currNorm.e().doubleValue(),B,result,lpSolver,A,x,y,8*backtrackTimes,1,5);//last para can be true or false. result.bestOptBasis==B
+										B = result.optBasis();
+										backtrackTimes++;
+										//if(DEBUG.debug) DEBUG.println(LPproject_DEBUG,"LPproject.findPlanes(): Let's try an new basis "+B.toString());
+									}						
+								}catch(NoSuchElementException e){
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**: While wrapping the caught exception into a custom one, information about the caught exception is being lost, including information about the stack trace of the exception.

**Fix**: If the caught exception object does not contain sensitive information, consider passing it as the &#34;rootCause&#34; or inner exception parameter to the constructor of the new exception before throwing the new exception. (Note that not all exception constructors support inner exceptions. Use a wrapper exception that supports inner exceptions.) 
[Learn more](https://www.ibm.com/support/pages/best-practice-catching-and-re-throwing-java-exceptions)